addons/account/models/product.py

```diff
@@ -118,6 +118,27 @@ def _check_uom_not_in_invoice(self):
                 "If you want to change its Unit of Measure, please archive this product and create a new one."
             ))
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        products = super().create(vals_list)
+        # If no company was set for the product, the product will be available for all companies and therefore should
+        # have the default taxes of the other companies as well. sudo() is used since we're going to need to fetch all
+        # the other companies default taxes which the user may not have access to.
+        other_companies = self.env['res.company'].sudo().search([('id', '!=', self.env.company.id)])
+        if not other_companies:
+            return products
+
+        default_customer_tax_ids = other_companies.filtered('account_sale_tax_id').account_sale_tax_id.ids
+        default_supplier_tax_ids = other_companies.filtered('account_purchase_tax_id').account_purchase_tax_id.ids
+
+        for product in products.sudo():
+            if not product.company_id:
+                if self.env.context.get('default_taxes_id', True):
```

Comment:
> Why are we checking the context?
> 
To have the same behaviour as the orm when it comes to `default_` context keys
> Note that the ORM team is working on removing the `default_` context keys
> 
I did not know that. Are they providing an alternative way then?
> Also, what happens if the tax values were set in the `vals`? Do we still want the default taxes on the other companies in this case?
>
Yes, since in the UI you're only seeing the taxes of the current company you're using, and if you're adding non default taxes they will stay the same for that company but for the rest of the companies, you'll get the default taxes.